### ISSUE TITLE
[Bug]: FaissException crash querynode and restart

### ISSUE THREAD

**Issue by psc0606** [milvus-io/milvus#18548]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.1
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.1, java-sdk-2.1.0-beta4
- OS(Ubuntu or CentOS): CentOS
- CPU/Memory: proxy(3 pods, 1core and 8G per pod), all coord(root, data, index) (1pod, 4core and 8G), datanode(10 pods, 2core and 8G per pod), indexnode(20 pods, 4core and 4G per pod), querynode(30pods, 1core and 8G per pod)
- GPU: None
- Others: None
```

### Current Behavior

I found the same issue[#17277 ]. But maybe the same error occur in milvus2.1.  FaissException made querynode crash and restart. I uploaded the log of the crashed querynode pod.

### Steps To Reproduce

```markdown
Sorry, I do know how to reproduce.
```

### Milvus Log

[milvus-prod-querynode-77f88b8776-8lwzt.log](https://github.com/milvus-io/milvus/files/9274461/milvus-prod-querynode-77f88b8776-8lwzt.log)

**Comment by soothing-rain** [comment]
/unassign
/assign @czs007 
Can you help look at this?

**Comment by yanliang567** [comment]
@cydrain is this the same issue that you are looking at? #18509 
/assing @cydrain 
/unassign @czs007

**Comment by cydrain** [comment]
from the log, this issue may be duplicated with #18509
```
2022-08-05T12:33:07+08:00	[2022/08/05 04:33:06.486 +00:00] [INFO] [querynode/reduce.go:155] ["purge memory done, memory was released back to the system successfully"] [usedMem=1107566592] [memPurgeRatio=0.2] [maxBinsSize=221513318]
2022-08-05T12:33:07+08:00	terminate called after throwing an instance of 'faiss::FaissException'
2022-08-05T12:33:07+08:00	terminate called recursively
2022-08-05T12:33:07+08:00	terminate called recursively
2022-08-05T12:33:07+08:00	terminate called recursively
```

**Comment by xiaofan-luan** [comment]
duplicated and should be fixed by 2.1.2
